materialize-elasticsearch/schemabuilder/src/elastic_search_schema_builder.rs

```diff
@@ -0,0 +1,427 @@
+use super::elastic_search_data_types::{ESBasicType, ESFieldType, ESTypeOverride};
+use super::errors::*;
+
+use doc::inference::{ArrayShape, ObjShape, Shape};
+use doc::Annotation;
+use flow_json::schema::{self, index::IndexBuilder, keywords, types};
+use serde_json::json;
+use serde_json::Value::{self, Object};
+use std::collections::HashMap;
+
+pub const DEFAULT_IGNORE_ABOVE: u16 = 256;
+
+pub fn build_elastic_schema(schema_json: &[u8]) -> Result<ESFieldType, Error> {
+    build_elastic_schema_with_overrides(schema_json, &[])
+}
+
+pub fn build_elastic_schema_with_overrides(
+    schema_json: &[u8],
+    es_type_overrides: &[ESTypeOverride],
+) -> Result<ESFieldType, Error> {
+    let schema: Value = serde_json::from_slice(schema_json)?;
+
+    let schema =
+        schema::build::build_schema::<Annotation>(get_schema_uri(&schema.clone())?, &schema)
+            .unwrap();
+
+    let mut index = IndexBuilder::new();
+    index.add(&schema).unwrap();
+    index.verify_references().unwrap();
```

Comment:
Got it. I was assuming the schema is passed in from CollectionSpec, which is expected to be perfect without any issues. However, I think it is better to propagate the error out so that we do not need to make any assumptions on the input schema.

I have converted `unwrap` here to `?`, and fixed the other places of `unwrap`.